src/options.rs

```diff
@@ -667,3 +693,26 @@ impl ReconnectDelayCallback {
         self.0(reconnects)
     }
 }
+
+pub(crate) struct ErrCallback(Option<Box<dyn Fn(Error) + Send + Sync + 'static>>);
```

Comment:
Nit (`Err` and `std::Error` are different things)

```suggestion
pub(crate) struct ErrorCallback(Option<Box<dyn Fn(Error) + Send + Sync + 'static>>);
```